### ISSUE TITLE
RFC: allow the creation of stopped Spans that can be started later manually.

### ISSUE THREAD

**Issue by t-8ch** [opentracing/opentracing-java#273]
This allows the span to be injected into a carrier before starting it.

The usecase is a buildsystem where the build environment is defined before the build is executed.
(Background info: https://groups.google.com/forum/#!topic/jenkinsci-dev/VBYvIv3S_r4)

**Comment by objectiser** [comment]
Why not create a span representing the "build environment is defined" and inject its context for use when the build is executed, and have another span created around the build execution defining a "follows from" reference back to the first span?

**Comment by yurishkuro** [comment]
I agree. I don't see anything special in this use case that cannot be done with the existing API. A pipeline of work units is naturally represented with a sequence of follows-from spans.

**Comment by t-8ch** [comment]
Unfortunately I do not control the execution inside the work units. Only of the scheduler.
This is not an urgent issue anymore, as a workaround in the setup logic of the scheduler has been implemented.
In general I still think functionality would be usable and hope it will end up on some wishlist for future development.

**Comment by yogeshVU** [comment]
In another use-case, I would like to start a span and stop a span and restart the span again without creating a new span id.  The use-case is a set of tasks that. run in parallel and finish executing at a given step than pauses, and restarts executing operation again in the next step. 

More info on the computation: https://en.wikipedia.org/wiki/Bulk_synchronous_parallel
